src/System.IO.Pipelines/PipeFactory.cs

```diff
@@ -49,9 +49,9 @@ public IPipeReader CreateReader(Stream stream)
             return pipe;
         }
 
-        private async void ExecuteCopyToAsync(Pipe pipe, Stream stream)
+        private Task ExecuteCopyToAsync(Pipe pipe, Stream stream)
         {
-            await stream.CopyToAsync(pipe);
```

Comment:
This now results in a CS4014 in CreateReader. This could cause a problem once we reenable warnings as errors which I'm sure will happen at some point. You can silence this with a pragma or `var ignore = ...`.

Aditionally, shouldn't there be a try/catch that sets an error on the pipe using IPipeWriter.Complete(ex) if CopyToAsync(pipe) throws? This would also prevent unobserved task execptions. Arguably, you wouldn't need to change this method to be Task returning if you did this, but I don't mind keeping it as async Task just in case Complete(ex) throws or something which it shouldn't